README.md
# Andrious Solutions Ltd.
## _Andrious Website_














> Andrious Solutions Ltd. extends our profound respect and immeasurable gratitude to all the ancestors and keepers of the land on whose traditional territories our work takes place. We acknowledge that we are on Treaty 1 territory, the traditional gathering place of the Anishinaabe, Cree, Oji-Cree, Dakota and Dene people and the traditional homeland of the Métis people. This land is sacred, historical, and significant.
> Every time we acknowledge this truth, we have an invitation and an opportunity to reflect on the wrongs of the past, what we do in the present, and what we can do to continually honour the people whose lands and water we benefit from today.
> This statement only acts as a first step in honouring the land we reside on and its peoples, and must be paired with education, understanding and informed action.

CHANGELOG.md

## 1.45.0+01
 November 27, 2022
- class WebPageWrapper
- interactiveViewer in Use Case Example on a small screen.
- Syntax and interface corrections

## 1.44.0
 April 27, 2022
- urlTap() function to launch App store websites

## 1.43.1+02
 April 20, 2022
- CNAME flutter.andrioussolutions.com
- device_preview: ^1.0.0
- firebase_options.dart

## 1.43.0+01
 April 16, 2022
- import 'package:ads/ads.dart' show Ads is commented out
- applicationId "com.andrioussolutions.bazaar_demo"
- msix_config to store on partner.microsoft.com
- Shrine & Bazaar apps are included.

## 1.42.0
 March 12, 2022
- Add robots.txt
- Add <noscript></noscript>

## 1.41.0
 March 10, 2022
- Introduce Company History
- SEO index.html

## 1.40.1
 March 02, 2022
- sdk: ">=2.16.1 <3.0.0" from ">=2.12.0 <3.0.0"
- Colors.white70 when in Dark mode.

## 1.40.0
 December 30, 2021
- routeInformationProvider: AppRouteInformationProvider(),

## 1.39.0
 December 28, 2021
- mvc_web 1.3.0

## 1.38.0
 December 17, 2021
- now using the package, mvc_web
- List<Widget>? child(BuildContext context, [WebPage? widget]) {

## 1.37.1
 December 13, 2021
- 'assets/images/use_case_example.jpg'
- this.reverse, in class WebPageBaseController

## 1.37.0
 December 13, 2021
- BasicScrollController
- Size get screenSize => _screenSize;
- MyApp.screenSize; to App.screenSize;
- T? webPageOf<T extends WebPage>(BuildContext context) =>
- _WebScrollbarState
- SingleChildScrollView( gets controller: scrollController,

## 1.36.0+03
 December 08, 2021
- mvc_application v7 to v8
- .git in build/
- .git in deployment/web/

## 1.35.0
 October 20, 2021
- child = RotatedBox(quarterTurns: 5, child: child);
- final void Function()? initState;

## 1.34.0
 October 20, 2021
- class FiveWhys extends WebPage {
- List<Widget> children04( renamed List<Widget> withHeader04(
- <div class="spin"></div>

## 1.33.0
 October 19, 2021
- class WebPageContainer extends StatelessWidget {

## 1.32.0+2
 October 17, 2021
- 'v. ${App.version}    Copyright © 2021 | Andrious Solutions Ltd.'

## 1.32.0
 October 17, 2021
- Disclosure: margin: EdgeInsets.fromLTRB(_screenSize.width * 0.1,

## 1.31.0
 October 17, 2021
- List<Widget> children05(BuildContext context, [WebPage? widget]) =>
- children.add(paradox.child(context, paradox)!);
- static late Size _screenSize;
- Removed icon: const Icon(Icons.share),

## 1.30.4
 October 15, 2021
-  _screenSize.width * 0.05,

## 1.30.3
 October 15, 2021
- _screenSize.width * (_smallScreen ? 0.05 :

## 1.30.2
 October 15, 2021
 - _screenSize.width * (_smallScreen ? 0.1 :

## 1.30.1
 October 15, 2021
- _screenSize.width * (_smallScreen ? 0.99 :

## 1.30.0
 October 15, 2021
- s1: 'GITTER',
- if (!_smallScreen && (_widget?.banner ?? false))

## 1.29.0
 October 14, 2021
- # - directives_ordering
- builder: builder ?? (BuildContext context) => const SizedBox(),
- padding: EdgeInsets.only(top: _screenSize.height * 0.2, bottom: 32),
- this.readMore = readMore;
- export 'package:flutter/gestures.dart'; // for TapGestureRecognizer()

## 1.28.1
 October 11, 2021
- _wrapper = widget as WebPageWrapper;
- Widget? child(BuildContext context, [WebPage? widget]) => _wrapper!._child ?? super.child(context)!;

## 1.28.0
 October 10, 2021
- Widget coverPage(BuildContext context, {void Function()? onTap
- final textStyle = Theme.of(context).textTheme.bodyText2;
- if (inSmallScreen) AppTheme.fontSizeButton,
- lightThemeData.copyWith(textTheme: customTheme.textTheme.copyWith(

## 1.27.0
 October 09, 2021
- Map<String, Object> toJson() =>
- AppRoutePath fromJson(Map<String, dynamic> json) {
- abstract class BasicController extends ControllerMVC {
- if (mounted) { setState(() {});

## 1.26.0
 October 04, 2021
- abstract class BasicController extends ControllerMVC {
- Removed void initWidget();
- if (topMargin == null || topMargin == double.nan || topMargin < 0) {

## 1.25.0
 October 03, 2021
- Flutter UI Examples
- class PrivacyPolicy extends WebPage<PrivacyPolicy> {
- StackWidgetProperties? screenOverlay(BuildContext context) => null;
- Widget popupScreen(
- @mustCallSuper
  void initWidget() => _controller.initWidget();
- MyApp._(Key? key) : super(key: key, errorReport: _onErrorReport);

## 1.24.0
 September 28, 2021
- PopupPage.window<void>( in initial_disclosure
- if (banner && !_smallScreen) to  if (banner)

## 1.23.0
 September 28, 2021
- class WebPage<T>
- class WebPageWrapper extends WebPage<WebPageWrapper>
- Material(child: InkWell(

## 1.22.0+2
 September 27, 2021
- _screenSize.height * (_smallScreen ? 0.01 : 0.35),

## 1.21.0
 September 27, 2021
- _screenSize.height * (_smallScreen ? 0 : 0.35),

## 1.20.0+2
 September 27, 2021
- child: InteractiveViewer(

## 1.19.0
 September 27, 2021
- MyApp.orientation(context);

## 1.18.0
 September 27, 2021
- padding: EdgeInsets.all(_smallScreen ? 0 : 40),

## 1.17.0
 September 27, 2021
- Fixed five_whys.png
- padding: const EdgeInsets.all(8),

## 1.16.0
 September 27, 2021
- physics: const NeverScrollableScrollPhysics()
- const String _threeTreesImage = 'assets/images/three_trees.jpg';
- const String _fiveWhysImage = 'assets/images/five_whys.png';

## 1.15.0
 September 27, 2021
- PopupPage.window<void>(context, ownScreen);

## 1.14.3+6
 September 26, 2021
- Remove Material from GridView

## 1.14.3+5
 September 26, 2021
- _margin = MyApp.screenSize.width * 0.2;

## 1.14.3+2
 September 26, 2021
- padding: EdgeInsets.only(top: 8, left: 8, bottom: 8, right: _right),

## 1.14.2+4
 September 26, 2021
- right: MyApp.inSmallScreen ? 90 : 8

## 1.14.2+3
 September 26, 2021
- shrinkWrap: MyApp.inSmallScreen,

## 1.14.2+2
 September 26, 2021
- SingleChildScrollView(primary: false, child: GridView.builder(

## 1.14.2
 September 25, 2021
- if (banner && !_smallScreen)

## 1.14.1+3
 September 25, 2021
- SizedBox(child: InteractiveViewer(

## 1.14.0
 September 25, 2021
- disclose = InitialDisclosure(banner: false);

## 1.13.0
 September 25, 2021
- fit: _smallScreen ? BoxFit.fill : BoxFit.cover,
- child: InteractiveViewer(

## 1.12.0

lib/main01.dart
// Copyright 2021 Andrious Solutions Ltd. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:andrious/src/view.dart';

void main() => runApp(MyApp());

class MyApp extends AppStatefulWidget {
  MyApp({Key? key}) : super(key: key);

  @override
  AppState createView() => AndriousWebsite();
}

class AndriousWebsite extends AppState {
  AndriousWebsite({Key? key})
      : super(
          key: key,
          debugShowCheckedModeBanner: false,
//          debugPaintSizeEnabled: true,
//          debugPaintPointersEnabled: true,
//          debugPaintLayerBordersEnabled: true,
          home: const MainPage(),
        );

  @override
  ThemeData onTheme() {
    //
    final TextTheme textTheme = TextTheme(
      headline1: GoogleFonts.ubuntu(
          fontSize: 98, fontWeight: FontWeight.w300, letterSpacing: -1.5),
      headline2: GoogleFonts.ubuntu(
          fontSize: 61, fontWeight: FontWeight.w300, letterSpacing: -0.5),
      headline3: GoogleFonts.ubuntu(fontSize: 49, fontWeight: FontWeight.w400),
      headline4: GoogleFonts.ubuntu(
          fontSize: 35, fontWeight: FontWeight.w400, letterSpacing: 0.25),
      headline5: GoogleFonts.ubuntu(fontSize: 24, fontWeight: FontWeight.w400),
      headline6: GoogleFonts.ubuntu(
          fontSize: 20, fontWeight: FontWeight.w500, letterSpacing: 0.15),
      subtitle1: GoogleFonts.ubuntu(
          fontSize: 16, fontWeight: FontWeight.w400, letterSpacing: 0.15),
      subtitle2: GoogleFonts.ubuntu(
          fontSize: 14, fontWeight: FontWeight.w500, letterSpacing: 0.1),
      bodyText1: GoogleFonts.cabin(
          fontSize: 17, fontWeight: FontWeight.w400, letterSpacing: 0.5),
      bodyText2: GoogleFonts.cabin(
          fontSize: 15, fontWeight: FontWeight.w400, letterSpacing: 0.25),
      button: GoogleFonts.cabin(
          fontSize: 15, fontWeight: FontWeight.w500, letterSpacing: 1.25),
      caption: GoogleFonts.cabin(
          fontSize: 13, fontWeight: FontWeight.w400, letterSpacing: 0.4),
      overline: GoogleFonts.cabin(
          fontSize: 11, fontWeight: FontWeight.w400, letterSpacing: 1.5),
    );

    const ColorScheme colorScheme = ColorScheme(
      primary: Color(0xFFFFFFFF),
      primaryVariant: Color(0xffB5BFD3),
      secondary: Color(0xffbb86fc),
      secondaryVariant: Color(0xffbb86fc),
      surface: Colors.black,
      background: Color(0xFFFFFFFF),
      error: Colors.red,
      onBackground: Colors.black,
      onError: Colors.red,
      onPrimary: Colors.black,
      onSecondary: Colors.black,
      onSurface: Colors.black,
      brightness: Brightness.light,
    );

    return ThemeData(
      brightness: colorScheme.brightness,
      primaryColor: colorScheme.primary,
      accentColor: colorScheme.secondary,
      errorColor: colorScheme.error,
      fontFamily: 'Cabin',
      textTheme: textTheme,
      primaryTextTheme: textTheme,
      accentTextTheme: textTheme,
      colorScheme: colorScheme,
      backgroundColor: colorScheme.background,
      textButtonTheme: TextButtonThemeData(
        style: TextButton.styleFrom(
          primary: colorScheme.secondary,
        ),
      ),
    );
  }
}
